server/src/main/java/org/opensearch/wlm/tracker/QueryGroupResourceUsageTrackerService.java

```diff
@@ -44,26 +44,21 @@ public QueryGroupResourceUsageTrackerService(TaskResourceTrackingService taskRes
      * @return Map of QueryGroup views
      */
     public Map<String, QueryGroupLevelResourceUsageView> constructQueryGroupLevelUsageViews() {
-        final Map<String, List<Task>> tasksByQueryGroup = getTasksGroupedByQueryGroup();
+        final Map<String, List<QueryGroupTask>> tasksByQueryGroup = getTasksGroupedByQueryGroup();
         final Map<String, QueryGroupLevelResourceUsageView> queryGroupViews = new HashMap<>();
 
         // Iterate over each QueryGroup entry
-        for (Map.Entry<String, List<Task>> queryGroupEntry : tasksByQueryGroup.entrySet()) {
-            // Compute the QueryGroup usage
-            final EnumMap<ResourceType, Long> queryGroupUsage = new EnumMap<>(ResourceType.class);
+        for (Map.Entry<String, List<QueryGroupTask>> queryGroupEntry : tasksByQueryGroup.entrySet()) {
+            // Compute the QueryGroup resource usage
+            final Map<ResourceType, Double> resourceUsage = new HashMap<>();
```

Comment:
nit: Any reason for changing from `EnumMap` to `HashMap`? Especially since `TRACKED_RESOURCES` is `EnumSet.allOf(ResourceType.class)`?